caikit/core/model_management/local_job_base.py

```diff
@@ -211,8 +212,14 @@ def wait(self):
         ## Impl ##
         def _delete_result(self):
             """Helper function to clear out the result when purging"""
-            if self.save_path and Path(self.save_path).exists():
-                Path(self.save_path).unlink(missing_ok=True)
+
+            if self.save_path:
+                save_pathlib = Path(self.save_path)
+                if save_pathlib.exists():
+                    if save_pathlib.is_file():
+                        save_pathlib.unlink(missing_ok=True)
+                    else:
+                        shutil.rmtree(save_pathlib, ignore_errors=True)
```

Comment:
🏌️ 

```suggestion
            if self.save_path and (save_pathlib := Path(self.save_path).exists():
                if save_pathlib.is_file():
                    save_pathlib.unlink(missing_ok=True)
                else:
                    shutil.rmtree(save_pathlib, ignore_errors=True)
```

![walrus](https://media.giphy.com/media/v1.Y2lkPTc5MGI3NjExc3p4endua25uYWt4emM5NWkwOHlxaHVndnFldzdmcnliaWR1bnVybiZlcD12MV9naWZzX3NlYXJjaCZjdD1n/3ohhwqVTyL61926M92/giphy.gif)